src/structurizr/model/element.py

```diff
@@ -119,7 +123,9 @@ def add_relationship(
                 f"Cannot add relationship {relationship} to element {self} that is not its source."
             )
         self.relationships.add(relationship)
-        self.get_model().add_relationship(relationship)
+        self.get_model().add_relationship(
```

Comment:
There is also a property if you prefer.

```suggestion
        self.model.add_relationship(
```